server/service/src/requisition_line/response_requisition_line/insert/generate.rs

```diff
@@ -0,0 +1,35 @@
+use repository::{RequisitionLineRow, RequisitionRow};
+
+use crate::{
+    requisition::request_requisition::generate_requisition_lines, service_provider::ServiceContext,
+};
+
+use super::{InsertResponseRequisitionLine, OutError};
+
+pub fn generate(
+    ctx: &ServiceContext,
+    store_id: &str,
+    requisition_row: RequisitionRow,
+    InsertResponseRequisitionLine {
+        id,
+        requisition_id: _,
+        item_id,
+        their_stock_on_hand,
+        requested_quantity,
+        supply_quantity,
+        comment,
+    }: InsertResponseRequisitionLine,
+) -> Result<RequisitionLineRow, OutError> {
+    let mut new_requisition_line =
+        generate_requisition_lines(ctx, store_id, &requisition_row, vec![item_id])?
+            .pop()
+            .ok_or(OutError::CannotFindItemStatusForRequisitionLine)?;
+
+    new_requisition_line.id = id;
+    new_requisition_line.initial_stock_on_hand_units = their_stock_on_hand.unwrap_or(0.0);
```

Comment:
Hmm I think this should be saved to `available_stock_on_hand`... and `initial_stock_on_hand_units`?

![image](https://github.com/user-attachments/assets/13c4c6cc-0b9e-41c6-bde3-7ecbe04446e2)

This is a response requisition generated from an internal order from a customer store. The current SOH is saved on `stock_on_hand` (which translates to `available_stock_on_hand`)

![image](https://github.com/user-attachments/assets/9b0a3e77-65cd-4928-88e1-40097eaffe57)

This is a manual response requisition with no extra columns. Seems to have saved to both `stock_on_hand` and `Cust_prev_stock_balance`. Wait wtf is the field `previous_stock_on_hand` isn't that `Cust_prev_stock_balance`?!

![image](https://github.com/user-attachments/assets/e43c826c-0880-46aa-a3fa-e8a2fc5e33a1)

OK nvm that field is basically not used lol

Reply:
I was going to say that it would overwrite the generated stock on hand for OMS customers, but then realised we get those from the linked requisition 👀 [e52d035](https://github.com/msupply-foundation/open-msupply/pull/5199/commits/e52d035357b1d8773750c1c15e39de6a214f9584)